### PR TITLE
chore: install same vitest version across packages

### PR DESCRIPTION
**Pull request by nflaig**
**Motivation**

Noticed when running `yarn install` locally on unstable branch it produces a lock file diff. This is due to the fact that we install different vitest versions.

I reviewed https://github.com/ChainSafe/lodestar/pull/6222 and seems to be the issue as the branch was open before we upgraded vitest in https://github.com/ChainSafe/lodestar/pull/6237. 

There is a way to enforce that a branch is rebased / merged against target branch before the PR can be merged. While I don't think this is strictly necessary, we should keep in mind to rebase long lived branches before merging.

**Description**

Install same vitest version across packages

### CODE REVIEW

File: packages/spec-test-util/src/single.ts [hunk 1]
```diff
@@ -112,7 +112,7 @@ export function describeDirectorySpecTest<TestCase extends {meta?: any}, Result>
         continue;
       }
 
-      // Use full path here, not just `testSubDirname` to allow usage of `vitest --grep`
+      // Use full path here, not just `testSubDirname` to allow usage of `vitest -t`
```
Comment (nflaig) [inline packages/spec-test-util/src/single.ts:115]: completely unrelated change, picked that up while reviewing https://github.com/ChainSafe/lodestar/pull/6222 and since this is a fixup of that PR, might as well include that here

**Review by wemeetagain — APPROVED** [review]

**Comment by wemeetagain** [comment]
:tada: This PR is included in [v1.15.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.15.0) :tada:
